File: QTD2.Infrastructure/Model/ILA/CBT/CBTCreateOptions.cs
using QTD2.Domain.Entities.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Infrastructure.Model.ILA.CBTReleaseSetting
{
    public class CBTCreateOptions
    {
        public bool CBTRequiredForCource { get; set; }
        public string CBTLearningContractInstructions { get; set; }
        public CBTAvailablity Availablity { get; set; }
        public int DueDateAmount { get; set; }
        public int EmpSettingsReleaseTypeId { get; set; }
    }
}

File: QTD.SPA/src/app/_DtoModels/ILA_CBT/CBTCreateOptions.ts
import { CBTAvailablity } from "./CBTAvailablity";
import { CBTDueDateInterval } from "./CBTDueDateInterval";

export class CBTCreateOptions{
    cBTRequiredForCource: boolean;
    cBTLearningContractInstructions: string;
    availablity: CBTAvailablity;
    dueDateAmount: number = 1;
    empSettingsReleaseTypeId :string;
    setCbtRequiredForCource(value:boolean){
        this.cBTRequiredForCource = value;
    }

    setCbtLearningInstructions(instructions: string){
        this.cBTLearningContractInstructions = instructions;
    }
    setCbtAvailability(value:CBTAvailablity){
        this.availablity = value;
    }

    setCbtDueDateAmount(dateAmount: number){
        this.dueDateAmount = dateAmount;
    }
    setEmpSettingsReleaseTypeId(dueDateTypeId: string){
        this.empSettingsReleaseTypeId = dueDateTypeId;
    }
}
